extension/content/tests/components/recipes/RecipeListing.test.tsx

```diff
@@ -0,0 +1,80 @@
+import { cleanup } from "@testing-library/react";
+import React from "react";
+
+import RecipeListing from "devtools/components/recipes/RecipeListing";
+
+import {
+  approvalRequestFactory,
+  revisionFactory,
+  recipeFactory,
+} from "devtools/tests/factories/recipes";
+
+afterEach(async () => {
+  jest.clearAllMocks();
+  await cleanup();
+});
+
+describe("RecipeListing", () => {
+  it("should have pending review tag when no approval", () => {
+    const approvalRequest = approvalRequestFactory.build({ approved: null });
+    const revision = revisionFactory.build(
+      { approval_request: approvalRequest },
+      { actionName: "branched-addon-study" },
+    );
+    const recipe = recipeFactory.build();
+    recipe.latest_revision = revision;
```

Comment:
I think you can collapse this down into a single call, since the new factories will pass details on through subfactories:

```suggestion
	const recipe = recipeFactory.build({
		latest_revision: {
			approval_request: {
				approved: null,
			},
		},
	});
```

Reply:
oooh did not know you could do that, will do